### PR TITLE
Add project

### PR DESCRIPTION
**Pull request by konrad-em**

### CODE REVIEW

File: BossManage/AddWorkerViewController.swift [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+//  AddWorkerViewController.swift
+//  BossManage
+//
+//  Created by Kacper Szufnarowski on 25.08.2018.
+//  Copyright © 2018 Kacper Szufnarowski. All rights reserved.
+//
+
+import UIKit
+
+class AddWorkerViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, InfoTableViewCellDelegate {
```
Comment (konrad-em) [inline BossManage/AddWorkerViewController.swift]: It looks much better if you keep it sectioned - keep class properties at the top, you can also create extension to conform to protocols and inside implement its methods

File: BossManage/AddWorkerViewController.swift [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    }
+    var name: String? = nil
+    var surname: String? = nil
+    weak var delegate: AddWorkerViewControlerDelegate?
+    @IBOutlet weak var NameField: UITextField! {
+        didSet {
+            NameField.attributedPlaceholder = NSAttributedString(string: "Name")
+        }
+    }
+    @IBOutlet weak var SurnameField: UITextField! {
```
Comment (konrad-em) [inline BossManage/AddWorkerViewController.swift]: Should not start with capital letter

File: BossManage/AddWorkerViewController.swift [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        cell?.editableText.textColor = UIColor.lightGray
+        return cell!
+    }
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return CGFloat(200)
+    }
+    var name: String? = nil
+    var surname: String? = nil
+    weak var delegate: AddWorkerViewControlerDelegate?
+    @IBOutlet weak var NameField: UITextField! {
```
Comment (konrad-em) [inline BossManage/AddWorkerViewController.swift]: Should not start with capital letter

File: BossManage/AddWorkerViewController.swift [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        cell?.delegate = self
+        cell?.infoAbout.text = infoTypes[indexPath.row]
+        cell?.editableText.text = "Add text..."
+        cell?.editableText.textColor = UIColor.lightGray
+        return cell!
+    }
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return CGFloat(200)
+    }
+    var name: String? = nil
```
Comment (konrad-em) [inline BossManage/AddWorkerViewController.swift]: No need to set is as nil here, it's optional defines it has no value until you set it

File: BossManage/AddWorkerViewController.swift [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "Info Cell", for: indexPath) as? InfoTableViewCell
+        cell?.delegate = self
+        cell?.infoAbout.text = infoTypes[indexPath.row]
+        cell?.editableText.text = "Add text..."
+        cell?.editableText.textColor = UIColor.lightGray
+        return cell!
+    }
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return CGFloat(200)
```
Comment (konrad-em) [inline BossManage/AddWorkerViewController.swift]: 200.0 is fine. Please consider 'tableView.rowHeight = UITableViewAutomaticDimension' when using auto layout

File: BossManage/AddPhotoView.swift [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+//  Copyright © 2018 Kacper Szufnarowski. All rights reserved.
+//
+
+import UIKit
+
+class AddPhotoView: UIView {
+
+    // Only override draw() if you perform custom drawing.
+    // An empty implementation adversely affects performance during animation.
+    var photoOfWorker = UIImage()
```
Comment (konrad-em) [inline BossManage/AddPhotoView.swift]: Does it need to be `var` ?

File: BossManage/AddWorkerViewController.swift [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+//
+//  Created by Kacper Szufnarowski on 25.08.2018.
+//  Copyright © 2018 Kacper Szufnarowski. All rights reserved.
+//
+
+import UIKit
+
+class AddWorkerViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, InfoTableViewCellDelegate {
+    func passString(passedString: String, fromCell: InfoTableViewCell) {
+        dataToSave.stringDataOfWorker[fromCell.infoAbout.text!] = passedString
```
Comment (konrad-em) [inline BossManage/AddWorkerViewController.swift]: Implicitly unwrapped optionals are not safe ;)

File: BossManage/AddWorkerViewController.swift [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return infoTypes.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "Info Cell", for: indexPath) as? InfoTableViewCell
+        cell?.delegate = self
+        cell?.infoAbout.text = infoTypes[indexPath.row]
+        cell?.editableText.text = "Add text..."
+        cell?.editableText.textColor = UIColor.lightGray
```
Comment (konrad-em) [inline BossManage/AddWorkerViewController.swift]: just be just `.lightGray`

File: BossManage/AddWorkerViewController.swift [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+        didSet {
+            NameField.attributedPlaceholder = NSAttributedString(string: "Name")
+        }
+    }
+    @IBOutlet weak var SurnameField: UITextField! {
+        didSet {
+            SurnameField.attributedPlaceholder = NSAttributedString(string: "Surname")
+        }
+    }
+    @IBOutlet weak var AddPhoto: AddPhotoView!
```
Comment (konrad-em) [inline BossManage/AddWorkerViewController.swift]: same here

File: BossManage/BossPersonTableViewController.swift [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return namesOfWorkres.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "Person", for: indexPath)
+        cell.textLabel?.text = namesOfWorkres[indexPath.row]
```
Comment (konrad-em) [inline BossManage/BossPersonTableViewController.swift]: not safe

File: BossManage/InfoTableViewCell.swift [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+        super.awakeFromNib()
+        editableText.delegate = self
+        let swipeGesture = UISwipeGestureRecognizer(target: self, action: #selector(self.resigning))
+        swipeGesture.direction = [.down]
+        self.addGestureRecognizer(swipeGesture)
+        // Initialization code
+    }
+    @IBOutlet weak var infoAbout: UILabel!
+    weak var delegate: InfoTableViewCellDelegate?
+    @objc func resigning() {
```
Comment (konrad-em) [inline BossManage/InfoTableViewCell.swift]: why `@objc` ?

Comment (Sumered) [reply BossManage/InfoTableViewCell.swift]: Need for calling selector at line 16

File: BossManage/ViewController.swift [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+//  ViewController.swift
+//  BossManage
+//
+//  Created by Kacper Szufnarowski on 24.08.2018.
+//  Copyright © 2018 Kacper Szufnarowski. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
```
[the remark this answers is not in the document]
Comment (Sumered) [reply BossManage/ViewController.swift]: It was leaved to be used in future.

**Pull request merged**